src/buildHooks.ts

```diff
@@ -67,7 +83,51 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
   mutationSelectors: MutationResultSelectors<Definitions, any>;
   mutationActions: MutationActions<Definitions>;
 }) {
-  return { buildQueryHook, buildMutationHook };
+  return { buildQueryHook, buildMutationHook, usePrefetch };
+
+  function usePrefetch<EndpointName extends QueryKeys<Definitions>>(
+    endpointName: EndpointName,
+    options?: PrefetchOptions
+  ) {
+    const dispatch = useDispatch<ThunkDispatch<any, any, AnyAction>>();
+    const store = useStore();
+    const latestStateRef = useRef(store.getState());
+
+    useLayoutEffect(() => {
+      return store.subscribe(() => {
+        latestStateRef.current = store.getState();
+      });
+    }, [store]);
+
+    return useCallback(
+      (arg: any, opts: PrefetchOptions = { force: false, ifOlderThan: false }) => {
+        const force = hasTheForce(opts) ? opts.force : hasTheForce(options) ? options.force : false;
+        const maxAge = hasMaxAge(opts) ? opts.ifOlderThan : hasMaxAge(options) ? options.ifOlderThan : false;
+
+        const queryAction = (force: boolean = true) => queryActions[endpointName](arg, { forceRefetch: force });
+
+        const latestStateValue = querySelectors[endpointName](arg)(latestStateRef.current);
+
+        if (force) {
+          dispatch(queryAction());
+        } else if (maxAge) {
+          const lastFulfilledTs = latestStateValue?.fulfilledTimeStamp;
+          if (!lastFulfilledTs) {
+            dispatch(queryAction());
+            return;
+          }
+          const shouldRetrigger = (Number(new Date()) - Number(new Date(lastFulfilledTs))) / 1000 >= maxAge;
+          if (shouldRetrigger) {
+            dispatch(queryAction());
+          }
+        } else {
+          // If prefetching with no options, just let it try
+          dispatch(queryAction(false));
+        }
+      },
+      [endpointName, dispatch, options]
```

Comment:
```suggestion
      [endpointName, dispatch, options, store]
```

Reply:
Skipping due to switching to prefetchThunk usage